### ISSUE TITLE
Move back to canonical isomorphic-git

### ISSUE THREAD

**Issue by crutchcorn** [oceanbit/GitShark#36]
Currently, we have migrated to using a fork of isomorphic-git with the build artifacts published to a branch. This is due to the lack of the following PRs being merged:

- https://github.com/isomorphic-git/isomorphic-git/pull/1194
- https://github.com/isomorphic-git/isomorphic-git/pull/1156

We're currently waiting on these being merged upstream before we're able to move back to stable

**Comment by chmac** [comment]
@crutchcorn The second PR of yours was merged, but unfortunately into `master` and not `main`, so I opened a new PR to copy that and 1 other change into `main`. https://github.com/isomorphic-git/isomorphic-git/pull/1270 Unfortunately it hasn't been merged yet.

As an aside, I'm using [patch-package](https://yarnpkg.com/package/patch-package) to fix the issue in my repo.

**Comment by crutchcorn** [comment]
@chmac I honestly have no idea how I missed this comment from you. I appreciate you opening a new PR to migrate those things for us!

FWIW, instead of using `patch-package` (which I've used for other scenarios), we've simply forked the repo and am using a copy from there:

https://github.com/oceanbit-dev/GitShark/blob/main/package.json#L43

Also worth noting: We've migrated away from `isomorphic-git` in favor of native code due to OOM issues we ran into

**Comment by crutchcorn** [comment]
I'm going to close this issue, as I'm realizing that migration to upstream `isomorphic-git` isn't realistic for our project and we'll likely just replace usage of `isomorphic-git` in our codebase entirely. 

That said, @chmac please don't feel like you can't comment more here (or, like I'd mentioned, joining one of those links to talk more in a more reflective real-time environment)
